docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,460 @@
+
+For example,
+
+```ini
+[Model] # Special section that describes model itself
+Type = NeuralNetwork  # Model Type : only NeuralNetwork is supported as of now
+Optimizer = adam  # Optimizer : sgd (stochastic gradient decent),
+          #             adam (Adaptive Moment Estimation)
+#### optimizer related properties
+Learning_rate = 0.0001  # Learning Rate for the optimizer
+Decay_rate = 0.96 # for the decay_rate for the decayed learning rate
+Decay_steps = 1000       # decay step for the exponential decayed learning rate
+beta1 = 0.9     # beta 1 for adam
+beta2 = 0.9999  # beta 2 for adam
+epsilon = 1e-7  # epsilon for adam
+
+#### Model compile related properties
+batch_size = 9
+loss = cross      # Cost(loss) function : mse (mean squared error)
+                        #                       cross (cross entropy)
+
+####  Model run related properties
+Epochs = 20     # Epochs
+save_path = "model.bin"   # model path to save / read parameters
+
+[DataSet] # Special section that describes dataset.
+BufferSize=9
+TrainData="trainingSet.dat"
+ValidData="validationSet.dat"
+LabelData="label.dat"
+
+# Layer Sections, each section name refers to name of the layer
+[inputlayer]
+Type = input
+Input_Shape = 1:1:62720 # Input Layer Dimension channel:height:width
+Normalization = true
+
+[outputlayer]
+Type = fully_connected
+Unit = 2    # Output Layer Dimension ( = Weight Width )
+bias_initializer = zeros
+weight_initializer = xavier_uniform
+Activation = sigmoid  # activation : sigmoid, softmax
+Weight_Decay = l2norm
+weight_Decay_Lambda = 0.005
+```
+
+Some restrictions must be kept:
+
+ * Model file must have a `[Model]` section.
+ * Model file must have at least one layer.
+ * Valid keys must have valid properties (while invalid keys in each section will be ignored.)
+
+Please note that all paths are relative to the current working directory unless stating an absolute path.
+It is recommended to set `save_path` and `Dataset` from the code over describing inside the model file.
+
+Following example constructs model from INI file
+
+```c
+char *res_path = app_get_resource_path();
+char model_path[1024];
+ml_train_model_h model;
+
+snprintf(model_path, sizeof(model_path), "%s/model.ini", res_path);
+free(res_path);
+
+status = ml_train_model_construct_with_conf(model_path, &model);
+if(status != ML_ERROR_NONE) {
+  // handle error
+}
+```
+
+### Construct a Model on the Code
+
+An empty model can be constructed with `ml_train_model_construct`.
+
+## Configure the Model
+
+After constructing a model, the model can be configured. This section provides example code to reproduce model description from [Create Model from INI Formatted File](#create-model-from-ini-formatted-file)
+
+Except that:
+ * Relative path is changed to dynamic app resource and data path.
+ * Model related properties (which can be set at compile / run phase)
+ * Demonstration about `ml_train_dataset_create_with_generator` which can't be covered in the description file.
+
```

Comment:
Please rephrase to use the Tizen writing style and tone.

Reply:
I skimmed through https://github.com/Samsung/tizen-docs/tree/master/styleguide but, I am not sure what defines `Tizen writing style and tone` . I rephrased a bit though. Could you elaborate more on this topic if modification isn't enought?